server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -81,12 +81,24 @@ public final class BulkRequestParser {
      * Create a new parser.
      *
      * @param deprecateOrErrorOnType whether to allow _type information in the index line; used by BulkMonitoring
+     * @param includeSourceOnError if to include the source in parser error messages
      * @param restApiVersion
      */
-    public BulkRequestParser(boolean deprecateOrErrorOnType, RestApiVersion restApiVersion) {
+    public BulkRequestParser(boolean deprecateOrErrorOnType, boolean includeSourceOnError, RestApiVersion restApiVersion) {
         this.deprecateOrErrorOnType = deprecateOrErrorOnType;
         this.config = XContentParserConfiguration.EMPTY.withDeprecationHandler(LoggingDeprecationHandler.INSTANCE)
-            .withRestApiVersion(restApiVersion);
+            .withRestApiVersion(restApiVersion)
+            .withIncludeSourceOnError(includeSourceOnError);
+    }
+
+    /**
+     * Create a new parser.
+     *
+     * @param deprecateOrErrorOnType whether to allow _type information in the index line; used by BulkMonitoring
+     * @param restApiVersion
+     */
+    public BulkRequestParser(boolean deprecateOrErrorOnType, RestApiVersion restApiVersion) {
```

Comment:
There aren't that many uses of this ctor (20ish). Can we update them to use thew 3 arg ctor so we don't need to maintain two similar ctors?